### DISCUSSION TITLE
Canonical Tautomer

### DISCUSSION THREAD

**Discussion by apahl** [rdkit/rdkit#6543] [Q&A] [answered]
I am sorry, but I am still in the giant rabbit hole that canonical tautomers turn out to be (see #6374).

Today I compared the output from the "old" canonicalizer (`Chem.MolStandardize.tautomer.TautomerCanonicalizer`) to the "new" one (from `Chem.MolStandardize.rdMolStandardize.TautomerEnumerator`) and was surprised to find that they give different results for some structures.  
An example:

![image](https://github.com/rdkit/rdkit/assets/9406836/fb1d34b5-c57f-4ccc-a80d-b929bf77d478)

Is this known behaviour?

(Circling back to the other discussion, I do like the structure from the "new" canonicalizer better)

KR Axel

**Comment by apahl** [comment]
This is with RDKit version ' 2023.03.1' on Ubuntu.

**Comment by greglandrum** [comment, marked as answer]
Hi @apahl. The rules for both tautomer enumeration and canonicalization in the C++ implementation (what you get from `rdMolStandardize`) have evolved from what was available in the MolVS python implementation (what you get from `tautomer`), so it's not unexpected that the results will be different for any particular molecule.

Unless you have a concrete reason for using the python implementation, which we are no longer working on, I'd suggest using the C++ one.

**Reply by apahl** [reply]
Thank you for the clarification, Greg.  
I was not aware of that, but ofc it makes sense.   
I will definitely use the C++ implementation for future datasets.

KR Axel
